build.gradle.kts

```diff
@@ -68,21 +65,18 @@ idea {
     }
 }
 
-val shadowed by configurations.creating
 configurations {
-    compileOnly {
-        extendsFrom(shadowed)
-    }
     testImplementation {
-        extendsFrom(shadowed)
         exclude(group = "junit", module = "junit")
     }
+    // Workaround https://github.com/gradle/gradle/issues/23928
+    shadow.configure { afterEvaluate { this@configure.dependencies.remove(project.dependencies.gradleApi()) } }
```

Comment:
I just verified, and this "workaround" does not really help. Is it required?
What helps is that the current `build.gradle.kts` configures manifest for `shadowJar` tasks, and it somehow removes `Class-Path` attribute.

Reply:
Huh? When I comment it out, the line appears, otherwise it's not there.

Reply:
> it somehow removes Class-Path attribute.

I think that might be because it throws away the `shadow` configuration in its entirety, fully reconfiguring the shadow plugin from scratch.

Reply:
Let's see https://github.com/gradle-nexus/publish-plugin/pull/168/commits/c14acc5dcaa8a9ff360d60aaf89f4965d109a189

Reply:
I downloaded `ubuntu-latest build folder.zip` (sha `ecc6e7f77e93cdc6e300595c1cfa33460b857079`)
`libs/publish-plugin-1.1.1-SNAPSHOT.jar` (sha `8ab95464dd7954b278ecf63338cb8ab55c3c4e47`)

Manifest contents is

```
Manifest-Version: 1.0
Implementation-Version: 1.1.1-SNAPSHOT
```

Reply:
I was a bit too bold with uploading the build/ folder, it contains the testkit folders 😅

https://github.com/gradle-nexus/publish-plugin/actions/runs/4204607452
(don't be scared about 952M, the compressed version is only 300M)

Anyway, `windows-latest build folder.zip/libs/publish-plugin-1.1.1-SNAPSHOT.jar` contains no .jars listed in the manifest, so the workaround works.

Reply:
I mean my local build (e.g. after removing `.gradle` and `build`) does not contain `class-path` attribute with the following diff:

```diff
diff --git a/build.gradle.kts b/build.gradle.kts
index 83ab07e..b96e667 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -70,7 +70,7 @@ configurations {
         exclude(group = "junit", module = "junit")
     }
     // Workaround https://github.com/gradle/gradle/issues/23928
-    shadow.configure { afterEvaluate { this@configure.dependencies.remove(project.dependencies.gradleApi()) } }
+//    shadow.configure { afterEvaluate { this@configure.dependencies.remove(project.dependencies.gradleApi()) } }
 }

 dependencies {
diff --git a/gradle/wrapper/gradle-wrapper.properties b/gradle/wrapper/gradle-wrapper.properties
index 2848974..42defcc 100644
--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-bin.zip
 networkTimeout=10000
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
```

Reply:
Attachments on https://github.com/gradle-nexus/publish-plugin/actions/runs/4206302177 show that the classpath is generated without the workaround.